src/Shared/Recruit.Vacancies.Client/Infrastructure/Providers/NextVacancyReviewProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Providers;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Esfa.Recruit.Vacancies.Client.Domain.Repositories;
+using Esfa.Recruit.Vacancies.Client.Domain.Services;
+
+namespace Esfa.Recruit.Vacancies.Client.Infrastructure.Providers
+{
+    public class NextVacancyReviewProvider : INextVacancyReviewProvider
```

Comment:
This sounds more like an Application Service i.e. working out what the next vacancy to look at is.

Reply:
Yeah. I flip-flopped on the naming of this.  As it returns a reviewId and doesn't change anything I landed on Provider. Happy to change. 👍

Reply:
I think it should be going by it definition. Means it will move from Infrastructure to Application.